packages/synapse-react-client/src/components/dataaccess/AccessRequirementDashboard.tsx

```diff
@@ -28,97 +30,84 @@ export function AccessRequirementDashboard(
   const { onCreateNewAccessRequirementClicked } = props
 
   const location = useLocation()
+  const urlSearchParams = useMemo(
+    () => new URLSearchParams(location.search),
+    [location.search],
+  )
   const history = useHistory()
 
-  const [nameOrID, setNameOrID] = useState<string>('')
-  const [relatedProjectId, setRelatedProjectId] = useState<string | undefined>(
-    undefined,
+  /**
+   * When an input changes, update the props passed to the table and update the search params.
+   *
+   * Debounced to prevent firing many queries while the user is entering text.
+   */
+  const [nameOrID, setNameOrID] = useDebouncedState<string>(
+    urlSearchParams.get(AR_NAME_OR_ID_SEARCH_PARAM_KEY) ?? '',
+    INPUT_CHANGE_DEBOUNCE_DELAY_MS,
+  )
+  const [relatedProjectId, setRelatedProjectId] = useDebouncedState<
+    string | undefined
+  >(
+    urlSearchParams.get(RELATED_PROJECT_ID_SEARCH_PARAM_KEY) ?? undefined,
+    INPUT_CHANGE_DEBOUNCE_DELAY_MS,
+  )
+  const [reviewerId, setReviewerId] = useDebouncedState<string | undefined>(
+    urlSearchParams.get(REVIEWER_ID_SEARCH_PARAM_KEY) ?? undefined,
+    INPUT_CHANGE_DEBOUNCE_DELAY_MS,
+  )
+  const [typeFilter, setTypeFilter] = useState<string | undefined>(
+    urlSearchParams.get(AR_TYPE_SEARCH_PARAM_KEY) ?? undefined,
   )
-  const [reviewerId, setReviewerId] = useState<string | undefined>(undefined)
 
   const projectFilterFieldIsError: boolean = useMemo(
     () =>
       !!(relatedProjectId && !SYNAPSE_ENTITY_ID_REGEX.exec(relatedProjectId)),
     [relatedProjectId],
   )
 
-  useEffect(() => {
-    function initializeFromSearchParams() {
-      const initialParams = new URLSearchParams(location.search)
-      setNameOrID(initialParams.get('nameOrID') ?? '')
-      setRelatedProjectId(initialParams.get('relatedProjectId') ?? undefined)
-      setReviewerId(initialParams.get('reviewerId') ?? undefined)
-    }
-    initializeFromSearchParams()
-  }, [location.search])
```

Comment:
Remove this effect and just initialize in the initial `useState`/`useDebouncedState` call